kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/IntOrString.java

```diff
@@ -58,153 +57,142 @@
-     * @return
-     *     The IntVal
-     */
-    @JsonProperty("IntVal")
-    public Integer getIntVal() {
-        return IntVal;
-    }
-
-    /**
-     *
-     * @param IntVal
-     *     The IntVal
-     */
-    @JsonProperty("IntVal")
-    public void setIntVal(Integer IntVal) {
-        this.IntVal = IntVal;
-    }
-
-    /**
-     *
-     * @return
-     *     The Kind
-     */
-    @JsonProperty("Kind")
-    public Integer getKind() {
-        return Kind;
-    }
-
-    /**
-     *
-     * @param Kind
-     *     The Kind
-     */
-    @JsonProperty("Kind")
-    public void setKind(Integer Kind) {
-        this.Kind = Kind;
-    }
-
-    /**
-     *
-     * @return
-     *     The StrVal
-     */
-    @JsonProperty("StrVal")
-    public String getStrVal() {
-        return StrVal;
-    }
-
-    /**
-     *
-     * @param StrVal
-     *     The StrVal
-     */
-    @JsonProperty("StrVal")
-    public void setStrVal(String StrVal) {
-        this.StrVal = StrVal;
-    }
-
-    @JsonAnyGetter
-    public Map<String, Object> getAdditionalProperties() {
-        return this.additionalProperties;
-    }
-
-    @JsonAnySetter
-    public void setAdditionalProperty(String name, Object value) {
-        this.additionalProperties.put(name, value);
-    }
-
-    public static class Serializer extends JsonSerializer<IntOrString> {
-
-        @Override
-        public void serialize(IntOrString value, JsonGenerator jgen, SerializerProvider provider) throws IOException, JsonProcessingException {
-            if (value != null) {
-                if (value.getKind() == null) {
-                    Integer intValue = value.getIntVal();
-                    if (intValue != null) {
-                        jgen.writeNumber(intValue);
-                    } else {
-                        String stringValue = value.getStrVal();
-                        if (stringValue != null) {
-                            jgen.writeString(stringValue);
-                        } else {
-                            jgen.writeNull();
-                        }
-                    }
-                } else if (value.getKind() == 0) {
-                    jgen.writeNumber(value.getIntVal());
-                } else if (value.getKind() == 1) {
-                    jgen.writeString(value.getStrVal());
-                } else {
-                    jgen.writeNull();
-                }
+  @JsonProperty("IntVal")
+  private Integer intVal;
+  @JsonProperty("Kind")
+  private Integer kind;
+  @JsonProperty("StrVal")
+  private String strVal;
+  @JsonIgnore
+  private Map<String, Object> additionalProperties = new HashMap<>();
+
+  public IntOrString() { }
+
+  //Builders are generated for the first non-empty constructor found.
+  public IntOrString(Integer intVal, Integer kind, String strVal, Map<String, Object> additionalProperties) {
+    this.intVal = intVal;
+    this.kind = kind;
+    this.strVal = strVal;
+    this.additionalProperties = additionalProperties;
+  }
```

Comment:
Maybe we don't want a builder for this.
We want to force users to use one of the provided constructors.